src/widget/emoticonswidget.h

```diff
@@ -31,12 +31,12 @@ class EmoticonsWidget : public QMenu
     explicit EmoticonsWidget(QWidget *parent = 0);
 
 signals:
-    void insertEmoticon(QString str);
+    void insertEmoticon(QString str) const;
 
 private slots:
-    void onSmileyClicked();
+    void onSmileyClicked() const;
     void onPageButtonClicked();
-    void PageButtonsUpdate();
+    void pageButtonsUpdate() const;
```

Comment:
Const slots don't make sense either, because they are literally meant to react to some event. If they don't change anything, their existence is questionable.